File: KuantumJAVA/KuantumNesnesi.java
public abstract class KuantumNesnesi {

    private String ID;
    private double stabilite;   // 0 - 100 arası
    private int tehlikeSeviyesi;  // 1 - 10 arası

    public KuantumNesnesi(String ID, double stabilite, int tehlikeSeviyesi) {
        this.ID = ID;
        setStabilite(stabilite);  // kapsülleme ile kontrol edilecek
        this.tehlikeSeviyesi = tehlikeSeviyesi;
    }

    public String getID() {
        return ID;
    }

    public double getStabilite() {
        return stabilite;
    }

    // KAPSÜLLEME
    public void setStabilite(double stabilite) {
        if (stabilite > 100) {
            this.stabilite = 100;
        } else if (stabilite < 0) {
            this.stabilite = 0;
        } else {
            this.stabilite = stabilite;
        }
    }

    public int getTehlikeSeviyesi() {
        return tehlikeSeviyesi;
    }

    // Abstract metot — çocuk sınıflar dolduracak
    public abstract void AnalizEt();

    // Tüm nesnelerde ortak
    public String DurumBilgisi() {
        return "ID: " + ID + " | Stabilite: " + stabilite;
    }
}

    

File: KuantumCS/KuantumNesnesi.cs
using System;

public abstract class KuantumNesnesi
{
    public string ID { get; set; }
    private int _stabilite;
    public int Stabilite 
    {
        get => _stabilite;
        set
        {
            if (value > 100) _stabilite = 100;
            else if (value < 0) _stabilite = 0;
            else _stabilite = value;
        }
    }

    public int TehlikeSeviyesi { get; set; }

    public KuantumNesnesi(string id, int stabilite, int tehlike)
    {
        ID = id;
        Stabilite = stabilite;
        TehlikeSeviyesi = tehlike;
    }

    public abstract void AnalizEt();

    public string DurumBilgisi()
    {
        return $"ID: {ID} | Stabilite: {Stabilite}";
    }
}
